src/APIBasedAuth.ts

```diff
@@ -132,42 +191,45 @@ class APIBasedAuth {
     }
 
     const { storage, storageKeys } = this.clientOptions;
-    for (const key of type === 'session' ? SESSION_KEYS : TOKEN_KEYS) {
-      const value = storage.removeItem(
-        storageKeys[key] || `${API_AUTH_STORAGE_KEY}.${key}`
-      );
-      if (value instanceof Promise) {
-        await value;
+    if (storage) {
+      for (const key of type === 'session' ? SESSION_KEYS : TOKEN_KEYS) {
+        await storage.removeItem(
+          storageKeys[key] || `${API_AUTH_STORAGE_KEY}.${key}`
+        );
       }
     }
   }
 
-  private async _store<T extends APIBasedAuth.StorageTypeName>(
-    type: T,
-    valuesToStore: APIBasedAuth.StorageType<T>
+  private async _store(
+    valuesToStore: APIBasedAuth.Session | APIBasedAuth.Tokens
   ) {
-    const { storage, storageKeys } = this.clientOptions;
-    if (type === 'session') {
-      this.session = valuesToStore as APIBasedAuth.Session;
+    if (valuesToStore._type === 'session') {
+      this.session = valuesToStore;
     }
 
-    for (const key of type === 'session' ? SESSION_KEYS : TOKEN_KEYS) {
-      const value = storage.setItem(
-        storageKeys[key] || `${API_AUTH_STORAGE_KEY}.${key}`,
-        valuesToStore[key]
-      );
-      if (value instanceof Promise) {
-        await value;
+    const { storage, storageKeys } = this.clientOptions;
+    if (storage) {
+      for (const key of valuesToStore._type === 'session'
+        ? SESSION_KEYS
+        : TOKEN_KEYS) {
+        await storage.setItem(
+          storageKeys[key] || `${API_AUTH_STORAGE_KEY}.${key}`,
+          valuesToStore[key]
+        );
       }
     }
   }
 }
 
 declare namespace APIBasedAuth {
   export type Config = {
+    /* LO clientId associated with API requests */
     clientId: string;
+    /* base URL for the endpoint used to make API requests */
     baseURL?: string;
+    /* interface for storage that can persist session/token values */
     storage?: Storage;
```

Comment:
Should this be made required now?

Reply:
yeah auth works without storage being set but there's no automatically handled token/session data to the user of app-tools if so. the user still has the ability to pull the values returned from using the functions and store them themselves